mopidy/core/tracklist.py

```diff
@@ -646,3 +647,25 @@ def _trigger_tracklist_changed(self):
     def _trigger_options_changed(self):
         logger.debug('Triggering options changed event')
         listener.CoreListener.send('options_changed')
+
+    def _save_state(self):
+        return TracklistState(
+            tl_tracks=self._tl_tracks,
+            next_tlid=self._next_tlid,
+            consume=self.get_consume(),
+            random=self.get_random(),
+            repeat=self.get_repeat(),
+            single=self.get_single())
+
+    def _load_state(self, state, coverage):
+        if state:
+            if 'mode' in coverage:
+                self.set_consume(state.consume)
+                self.set_random(state.random)
+                self.set_repeat(state.repeat)
+                self.set_single(state.single)
+            if 'tracklist' in coverage:
+                self._next_tlid = max(state.next_tlid, self._next_tlid)
+                self._tl_tracks = []
+                self._tl_tracks.extend(state.tl_tracks)
```

Comment:
`self._tl_tracks = list(state.tl_tracks)` perhaps, assuming `tl_tracks` is a tuple in the state model?